lib/graphql_metrics/extractor.rb

```diff
@@ -133,17 +142,19 @@ def extract_batch_loaders
     def extract_query
       return unless respond_to?(:query_extracted)
 
-      start_time = ctx_namespace[START_TIME_KEY]
-      return unless start_time
+      start_time = ctx_namespace[START_TIME_KEY] if ctx_namespace
```

Comment:
maybe two separate returns?

```ruby
return unless ctx_namespace
return unless (start_time = ctx_namespace[START_TIME_KEY])
```

Reply:
That would change the behaviour away from `return unless start_time || ctx_namespace.nil?`

I want to return early if the code is running as instrumentation, and we don't have a start_time, but don't want to return early when we're running `extract!` (and `ctx_namespace` isn't set). I may make this more explicit, and introduce a concept of runtime modes: one-off (extract!) vs. instrumentation, and check that instead of related values.